pysam/libcalignmentfile.pyx

```diff
@@ -1568,15 +1569,15 @@ cdef class AlignmentFile(HTSFile):
             seq = read.seq
             quality = read.query_qualities
             
-            not_check_qual = 0
-            if not quality:
-                warnings.warn('%s contains QUAL field, all bases were counted' %read.query_name)
-                not_check_qual = 1
+            if not quality and check_qual == 1:
+                raise ValueError('%s contains no QUAL field, use quality_threshold = None to ignore base quality' %read.query_name)
```

Comment:
I do not recommend raising an error here.  If quality is not set, then it should fail the quality threshold check.

Reply:
The reason I added this is that the error raised from failing the quality threshold check isn't so intuitive, since it is just an ```TypeError: 'NoneType' object has no attribute '__getitem__'```  from failing to excute ```quality[qpos]```. Is there a good way to fix this?

Reply:
I'd write the check to be:
```python
    if _threshold and (not quality or quality[qpos] < _threshold):
```